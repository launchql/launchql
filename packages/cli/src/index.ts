#!/usr/bin/env node
import { CLI, CLIOptions } from 'inquirerer';

import { commands } from './commands';

export const options: Partial<CLIOptions> = {
  minimistOpts: {
    alias: {
      v: 'version',
      h: 'help'
    }
  }
};

const app = new CLI(commands, options);

app.run().then(()=> {
  // all done!
}).catch(error => {
  // Should not reach here with the new CLI error handling pattern
  // But keep as fallback for unexpected errors
  console.error('Unexpected error:', error);
  process.exit(1);
});