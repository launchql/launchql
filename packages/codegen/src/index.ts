import { generateCode } from './codegen';
import getIntrospectionRows, { GetIntrospectionRowsOptions } from './introspect';
import { DatabaseObject } from './types';

(async () => {
  const options: GetIntrospectionRowsOptions = {
    connectionString: 'postgresql://postgres:password@localhost:5432/mydb',
    introspectionOptions: {
      pgLegacyFunctionsOnly: false,
      pgIgnoreRBAC: true,
    },
    namespacesToIntrospect: ['collections_public'],
    includeExtensions: false,
  };

  try {
    // Fetch introspection rows
    const rows: DatabaseObject[] = await getIntrospectionRows(options);
    console.log('Introspection Rows Fetched:', rows);

    // Generate TypeScript code
    const codegenOptions = {
      includeTimestamps: true,
      includeUUID: true,
    };
    const generatedCode = generateCode(rows, codegenOptions);

    console.log('Generated TypeScript Code:');
    console.log(generatedCode);
  } catch (error) {
    console.error('Failed to fetch introspection rows or generate code:', error);
  }
})();
