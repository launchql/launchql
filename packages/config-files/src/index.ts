export * from './types';
export * from './plan';
export * from './config';
export * from './sql';
export * from './extension';
export * from './sql-scripts';