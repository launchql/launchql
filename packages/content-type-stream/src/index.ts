export * from './content-stream';
export * from './content-type-stream';