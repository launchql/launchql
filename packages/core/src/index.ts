export * from './core/class/launchql';
export * from './export/export-meta';
export * from './export/export-migrations';
export * from './extensions/extensions';
export * from './modules/modules';
export * from './packaging/package';
export * from './packaging/transform';
export * from './resolution/deps';
export * from './resolution/resolve';
export * from './workspace/paths';
export * from './workspace/utils';

// Export project-files functionality (now integrated into core)
export * from './files';
export { cleanSql } from './migrate/clean';
export { LaunchQLMigrate } from './migrate/client';
export { 
  DeployOptions, 
  DeployResult, 
  MigrateChange, 
  MigratePlanFile, 
  RevertOptions, 
  RevertResult, 
  StatusResult,
  VerifyOptions, 
  VerifyResult} from './migrate/types';
export { hashFile, hashString } from './migrate/utils/hash';
export { executeQuery,TransactionContext, TransactionOptions, withTransaction } from './migrate/utils/transaction';
