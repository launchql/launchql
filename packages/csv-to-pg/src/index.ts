// @ts-nocheck
export * from './parse';
export * from './parser';
