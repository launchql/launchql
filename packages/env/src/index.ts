export { getEnvOptions, getConnEnvOptions, getDeploymentEnvOptions } from './merge';
export { loadConfigSync, loadConfigSyncFromDir } from './config';
export { getEnvVars } from './env';
export { walkUp, sluggify } from './utils';

export type { LaunchQLOptions, PgTestConnectionOptions, DeploymentOptions } from '@launchql/types';
