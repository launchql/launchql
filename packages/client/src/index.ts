import { Pool, PoolClient } from 'pg';

import env from './env';

export class Database {
  private static instance: Database;
  private pool: Pool;

  constructor() {
    if (Database.instance) {
      return Database.instance;
    }

    const pgPoolConfig = {
      connectionString: env.DATABASE_URL,
    };
    this.pool = new Pool(pgPoolConfig);

    // Ensure the pool is closed on process termination
    process.on('SIGTERM', async () => {
      await this.shutdown();
    });

    Database.instance = this;
    return this;
  }

  /**
   * Executes a callback function within a database transaction.
   * @param fn - A callback function that receives a PoolClient to perform database operations.
   */
  async withTransaction(fn: (client: PoolClient) => Promise<void>): Promise<void> {
    const client = await this.pool.connect();
    try {
      await client.query('BEGIN');
      try {
        await fn(client);
        await client.query('COMMIT');
      } catch (e) {
        console.error('Error during transaction:', e);
        await client.query('ROLLBACK');
        throw e; // Re-throw the error to propagate it
      }
    } finally {
      client.release();
    }
  }

  /**
   * Shuts down the connection pool.
   */
  async shutdown(): Promise<void> {
    await this.pool.end();
  }
}