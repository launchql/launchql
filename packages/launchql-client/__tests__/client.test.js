import introspection from '../__fixtures__/api/introspection.json';
import metaObject from '../__fixtures__/api/meta-obj.json';
import { Client } from '../src';

describe('getMany', () => {
  it('should select only scalar fields by default', () => {
    const client = new Client({
      meta: metaObject,
      introspection
    });

    const result = client.query('Action').getMany().print();

    expect(result._hash).toMatchSnapshot();
    expect(result._queryName).toMatchSnapshot();
    expect(
      /(ownerId)|(userActions)|(userActionResults)|(userActionItems)/.test(
        result._hash
      )
    ).toBe(false);
  });

  it('should whitelist selected fields', () => {
    const client = new Client({
      meta: metaObject,
      introspection
    });

    const result = client
      .query('Action')
      .getMany({
        select: {
          id: true,
          name: true,
          photo: true,
          title: true,
          actionResults: {
            select: {
              id: true,
              actionId: true
            },
            variables: {
              first: 10,
              filter: {
                name: {
                  in: ['abc', 'def']
                },
                actionId: { equalTo: 'dc310161-7a42-4b93-6a56-9fa48adcad7e' }
              }
            }
          }
        }
      })
      .print();

    expect(result._hash).toMatchSnapshot();
    expect(result._queryName).toMatchSnapshot();
  });
});

it('should select totalCount in subfields by default', () => {
  const client = new Client({
    meta: metaObject,
    introspection
  });

  const result = client
    .query('Action')
    .getMany({
      select: {
        id: true,
        name: true,
        photo: true,
        title: true,
        actionResults: {
          select: {
            id: true,
            actionId: true
          },
          variables: {
            first: 10,
            filter: {
              name: {
                in: ['abc', 'def']
              },
              actionId: { equalTo: 'dc310161-7a42-4b93-6a56-9fa48adcad7e' }
            }
          }
        }
      }
    })
    .print();

  expect(/(totalCount)/.test(result._hash)).toBe(true);
  expect(result._hash).toMatchSnapshot();
  expect(result._queryName).toMatchSnapshot();
});

it('selects relation field', () => {
  const client = new Client({
    meta: metaObject,
    introspection
  });

  const result = client
    .query('Action')
    .getMany({
      select: {
        id: true,
        slug: true,
        photo: true,
        title: true,
        url: true,
        goals: {
          select: {
            id: true,
            name: true,
            slug: true,
            shortName: true,
            icon: true,
            subHead: true,
            tags: true,
            search: true,
            createdBy: true,
            updatedBy: true,
            createdAt: true,
            updatedAt: true
          },
          variables: {
            first: 3
          }
        }
      }
    })
    .print();

  expect(/(goals)/.test(result._hash)).toBe(true);
  expect(result._hash).toMatchSnapshot();
});

it('selects all scalar fields of junction table by default', () => {
  const client = new Client({
    meta: metaObject,
    introspection
  });

  const result = client.query('ActionGoal').getMany().print();

  expect(/(actionId)|(goalId)|(ownerId)/.test(result._hash)).toBe(true);
  expect(result._hash).toMatchSnapshot();
});

it('selects belongsTo relation field', () => {
  const client = new Client({
    meta: metaObject,
    introspection
  });

  const result = client
    .query('Action')
    .getMany({
      select: {
        owner: {
          select: {
            id: true,
            type: true
          }
        }
      }
    })
    .print();

  // Straight up select the sub fields
  expect(/(owner)|(id)|(type)/gm.test(result._hash)).toBe(true);
  expect(result._hash).toMatchSnapshot();
});

it('selects non-scalar custom types', () => {
  const client = new Client({
    meta: metaObject,
    introspection
  });

  const result = client
    .query('Action')
    .getMany({
      select: {
        id: true,
        name: true,
        location: true, // non-scalar custom type
        timeRequired: true // non-scalar custom type
      }
    })
    .print();

  expect(/(totalCount)/.test(result._hash)).toBe(true);
  expect(result._queryName).toMatchSnapshot();
});

it('getMany edges', () => {
  const client = new Client({
    meta: metaObject,
    introspection
  });
  const result = client
    .query('Action')
    .edges(true)
    .getMany({
      select: {
        id: true,
        name: true,
        photo: true,
        title: true,
        actionResults: {
          select: {
            id: true,
            actionId: true
          },
          variables: {
            first: 10,
            before: null,
            filter: {
              name: {
                in: ['abc', 'def']
              },
              actionId: { equalTo: 'dc310161-7a42-4b93-6a56-9fa48adcad7e' }
            }
          }
        }
      }
    })
    .print();
  expect(result._hash).toMatchSnapshot();
  expect(result._queryName).toMatchSnapshot();
});

it('getOne', () => {
  const client = new Client({
    meta: metaObject,
    introspection
  });
  const result = client
    .query('Action')
    .getOne({
      select: {
        id: true,
        name: true,
        photo: true,
        title: true,
        actionResults: {
          select: {
            id: true,
            actionId: true
          },
          variables: {
            first: 10,
            before: null,
            filter: {
              name: {
                in: ['abc', 'def']
              },
              actionId: { equalTo: 'dc310161-7a42-4b93-6a56-9fa48adcad7e' }
            }
          }
        }
      }
    })
    .print();
  expect(result._hash).toMatchSnapshot();
  expect(result._queryName).toMatchSnapshot();
});

it('getAll', () => {
  const client = new Client({
    meta: metaObject,
    introspection
  });

  const result = client
    .query('Action')
    .all({
      select: {
        id: true,
        name: true,
        photo: true,
        title: true,
        actionResults: {
          select: {
            id: true,
            actionId: true
          },
          variables: {
            first: 10,
            before: null,
            filter: {
              name: {
                in: ['abc', 'def']
              },
              actionId: { equalTo: 'dc310161-7a42-4b93-6a56-9fa48adcad7e' }
            }
          }
        }
      }
    })
    .print();
  expect(result._hash).toMatchSnapshot();
  expect(result._queryName).toMatchSnapshot();
});

it('create with default scalar selection', () => {
  const client = new Client({
    meta: metaObject,
    introspection
  });
  const result = client.query('Action').create().print();

  expect(result._hash).toMatchSnapshot();
  expect(result._queryName).toMatchSnapshot();
});

it('create with custom selection', () => {
  const client = new Client({
    meta: metaObject,
    introspection
  });
  const result = client
    .query('Action')
    .create({
      select: {
        id: true,
        name: true,
        photo: true,
        title: true
      }
    })
    .print();

  expect(/(id)|(name)|(photo)|(title)/gm.test(result._hash)).toBe(true);
  expect(result._hash).toMatchSnapshot();
  expect(result._queryName).toMatchSnapshot();
});

it('update with default scalar selection', () => {
  const client = new Client({
    meta: metaObject,
    introspection
  });
  const result = client.query('Action').update().print();
  expect(result._hash).toMatchSnapshot();
  expect(result._queryName).toMatchSnapshot();
});

it('update with custom selection', () => {
  const client = new Client({
    meta: metaObject,
    introspection
  });
  const result = client
    .query('Action')
    .update({
      select: {
        id: true,
        name: true,
        photo: true,
        title: true
      }
    })
    .print();
  expect(/(id)|(name)|(photo)|(title)/gm.test(result._hash)).toBe(true);
  expect(result._hash).toMatchSnapshot();
  expect(result._queryName).toMatchSnapshot();
});

it('delete', () => {
  const client = new Client({
    meta: metaObject,
    introspection
  });
  const result = client.query('Action').delete().print();
  expect(result._hash).toMatchSnapshot();
  expect(result._queryName).toMatchSnapshot();
});

it('expands further selections of custom ast fields in nested selection', () => {
  const client = new Client({
    meta: metaObject,
    introspection
  });

  const result = client
    .query('ActionGoal')
    .getMany({
      select: {
        action: {
          select: {
            id: true,
            location: true, // custom ast
            timeRequired: true // custom ast
          }
        }
      }
    })
    .print();

  expect(result._hash).toMatchSnapshot();
  expect(result._queryName).toMatchSnapshot();
  expect(
    /(geojson)|(days)|(hours)|(minutes)|(months)|(seconds)|(years)/.test(
      result._hash
    )
  ).toBe(true);
});
