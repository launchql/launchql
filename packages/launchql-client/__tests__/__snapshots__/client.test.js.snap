// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create with custom selection 1`] = `
"mutation createActionMutation($id: UUID, $slug: String, $photo: JSON, $shareImage: JSON, $title: String, $titleObjectTemplate: String, $url: String, $description: String, $discoveryHeader: String, $discoveryDescription: String, $notificationText: String, $notificationObjectTemplate: String, $enableNotifications: Boolean, $enableNotificationsText: String, $search: FullText, $location: GeoJSON, $locationRadius: BigFloat, $timeRequired: IntervalInput, $startDate: Datetime, $endDate: Datetime, $approved: Boolean, $published: Boolean, $isPrivate: Boolean, $rewardAmount: BigFloat, $activityFeedText: String, $callToAction: String, $completedActionText: String, $alreadyCompletedActionText: String, $selfVerifiable: Boolean, $isRecurring: Boolean, $recurringInterval: IntervalInput, $oncePerObject: Boolean, $minimumGroupMembers: Int, $limitedToLocation: Boolean, $tags: [String], $groupId: UUID, $ownerId: UUID!, $objectTypeId: Int, $rewardId: UUID, $verifyRewardId: UUID, $photoUpload: Upload, $shareImageUpload: Upload) {
  createAction(input: {action: {id: $id, slug: $slug, photo: $photo, shareImage: $shareImage, title: $title, titleObjectTemplate: $titleObjectTemplate, url: $url, description: $description, discoveryHeader: $discoveryHeader, discoveryDescription: $discoveryDescription, notificationText: $notificationText, notificationObjectTemplate: $notificationObjectTemplate, enableNotifications: $enableNotifications, enableNotificationsText: $enableNotificationsText, search: $search, location: $location, locationRadius: $locationRadius, timeRequired: $timeRequired, startDate: $startDate, endDate: $endDate, approved: $approved, published: $published, isPrivate: $isPrivate, rewardAmount: $rewardAmount, activityFeedText: $activityFeedText, callToAction: $callToAction, completedActionText: $completedActionText, alreadyCompletedActionText: $alreadyCompletedActionText, selfVerifiable: $selfVerifiable, isRecurring: $isRecurring, recurringInterval: $recurringInterval, oncePerObject: $oncePerObject, minimumGroupMembers: $minimumGroupMembers, limitedToLocation: $limitedToLocation, tags: $tags, groupId: $groupId, ownerId: $ownerId, objectTypeId: $objectTypeId, rewardId: $rewardId, verifyRewardId: $verifyRewardId, photoUpload: $photoUpload, shareImageUpload: $shareImageUpload}}) {
    action {
      id
      photo
      title
    }
  }
}
"
`;

exports[`create with custom selection 2`] = `"createActionMutation"`;

exports[`create with default scalar selection 1`] = `
"mutation createActionMutation($id: UUID, $slug: String, $photo: JSON, $shareImage: JSON, $title: String, $titleObjectTemplate: String, $url: String, $description: String, $discoveryHeader: String, $discoveryDescription: String, $notificationText: String, $notificationObjectTemplate: String, $enableNotifications: Boolean, $enableNotificationsText: String, $search: FullText, $location: GeoJSON, $locationRadius: BigFloat, $timeRequired: IntervalInput, $startDate: Datetime, $endDate: Datetime, $approved: Boolean, $published: Boolean, $isPrivate: Boolean, $rewardAmount: BigFloat, $activityFeedText: String, $callToAction: String, $completedActionText: String, $alreadyCompletedActionText: String, $selfVerifiable: Boolean, $isRecurring: Boolean, $recurringInterval: IntervalInput, $oncePerObject: Boolean, $minimumGroupMembers: Int, $limitedToLocation: Boolean, $tags: [String], $groupId: UUID, $ownerId: UUID!, $objectTypeId: Int, $rewardId: UUID, $verifyRewardId: UUID, $photoUpload: Upload, $shareImageUpload: Upload) {
  createAction(input: {action: {id: $id, slug: $slug, photo: $photo, shareImage: $shareImage, title: $title, titleObjectTemplate: $titleObjectTemplate, url: $url, description: $description, discoveryHeader: $discoveryHeader, discoveryDescription: $discoveryDescription, notificationText: $notificationText, notificationObjectTemplate: $notificationObjectTemplate, enableNotifications: $enableNotifications, enableNotificationsText: $enableNotificationsText, search: $search, location: $location, locationRadius: $locationRadius, timeRequired: $timeRequired, startDate: $startDate, endDate: $endDate, approved: $approved, published: $published, isPrivate: $isPrivate, rewardAmount: $rewardAmount, activityFeedText: $activityFeedText, callToAction: $callToAction, completedActionText: $completedActionText, alreadyCompletedActionText: $alreadyCompletedActionText, selfVerifiable: $selfVerifiable, isRecurring: $isRecurring, recurringInterval: $recurringInterval, oncePerObject: $oncePerObject, minimumGroupMembers: $minimumGroupMembers, limitedToLocation: $limitedToLocation, tags: $tags, groupId: $groupId, ownerId: $ownerId, objectTypeId: $objectTypeId, rewardId: $rewardId, verifyRewardId: $verifyRewardId, photoUpload: $photoUpload, shareImageUpload: $shareImageUpload}}) {
    action {
      id
      slug
      photo
      shareImage
      title
      titleObjectTemplate
      url
      description
      discoveryHeader
      discoveryDescription
      notificationText
      notificationObjectTemplate
      enableNotifications
      enableNotificationsText
      search
      location {
        geojson
      }
      locationRadius
      timeRequired {
        days
        hours
        minutes
        months
        seconds
        years
      }
      startDate
      endDate
      approved
      published
      isPrivate
      rewardAmount
      activityFeedText
      callToAction
      completedActionText
      alreadyCompletedActionText
      selfVerifiable
      isRecurring
      recurringInterval {
        days
        hours
        minutes
        months
        seconds
        years
      }
      oncePerObject
      minimumGroupMembers
      limitedToLocation
      tags
      createdBy
      updatedBy
      createdAt
      updatedAt
    }
  }
}
"
`;

exports[`create with default scalar selection 2`] = `"createActionMutation"`;

exports[`delete 1`] = `
"mutation deleteActionMutation($id: UUID!) {
  deleteAction(input: {id: $id}) {
    clientMutationId
  }
}
"
`;

exports[`delete 2`] = `"deleteActionMutation"`;

exports[`expands further selections of custom ast fields in nested selection 1`] = `
"query getActionGoalsQuery($first: Int, $last: Int, $after: Cursor, $before: Cursor, $offset: Int, $condition: ActionGoalCondition, $filter: ActionGoalFilter, $orderBy: [ActionGoalsOrderBy!]) {
  actionGoals(first: $first, last: $last, offset: $offset, after: $after, before: $before, condition: $condition, filter: $filter, orderBy: $orderBy) {totalCount, pageInfo {
    hasNextPage
    hasPreviousPage
    endCursor
    startCursor
  }, nodes {
    action {
      id
      location {
        geojson
      }
      timeRequired {
        days
        hours
        minutes
        months
        seconds
        years
      }
    }
  }}
}
"
`;

exports[`expands further selections of custom ast fields in nested selection 2`] = `"getActionGoalsQuery"`;

exports[`getAll 1`] = `
"query getActionsQueryAll {
  actions {totalCount, nodes {
    id
    photo
    title
  }}
}
"
`;

exports[`getAll 2`] = `"getActionsQueryAll"`;

exports[`getMany edges 1`] = `
"query getActionsQuery($first: Int, $last: Int, $after: Cursor, $before: Cursor, $offset: Int, $condition: ActionCondition, $filter: ActionFilter, $orderBy: [ActionsOrderBy!]) {
  actions(first: $first, last: $last, offset: $offset, after: $after, before: $before, condition: $condition, filter: $filter, orderBy: $orderBy) {totalCount, pageInfo {
    hasNextPage
    hasPreviousPage
    endCursor
    startCursor
  }, edges {
    cursor
    node {
      id
      photo
      title
    }
  }}
}
"
`;

exports[`getMany edges 2`] = `"getActionsQuery"`;

exports[`getMany should select only scalar fields by default 1`] = `
"query getActionsQuery($first: Int, $last: Int, $after: Cursor, $before: Cursor, $offset: Int, $condition: ActionCondition, $filter: ActionFilter, $orderBy: [ActionsOrderBy!]) {
  actions(first: $first, last: $last, offset: $offset, after: $after, before: $before, condition: $condition, filter: $filter, orderBy: $orderBy) {totalCount, pageInfo {
    hasNextPage
    hasPreviousPage
    endCursor
    startCursor
  }, nodes {
    id
    slug
    photo
    shareImage
    title
    titleObjectTemplate
    url
    description
    discoveryHeader
    discoveryDescription
    notificationText
    notificationObjectTemplate
    enableNotifications
    enableNotificationsText
    search
    location {
      geojson
    }
    locationRadius
    timeRequired {
      days
      hours
      minutes
      months
      seconds
      years
    }
    startDate
    endDate
    approved
    published
    isPrivate
    rewardAmount
    activityFeedText
    callToAction
    completedActionText
    alreadyCompletedActionText
    selfVerifiable
    isRecurring
    recurringInterval {
      days
      hours
      minutes
      months
      seconds
      years
    }
    oncePerObject
    minimumGroupMembers
    limitedToLocation
    tags
    createdBy
    updatedBy
    createdAt
    updatedAt
  }}
}
"
`;

exports[`getMany should select only scalar fields by default 2`] = `"getActionsQuery"`;

exports[`getMany should whitelist selected fields 1`] = `
"query getActionsQuery($first: Int, $last: Int, $after: Cursor, $before: Cursor, $offset: Int, $condition: ActionCondition, $filter: ActionFilter, $orderBy: [ActionsOrderBy!]) {
  actions(first: $first, last: $last, offset: $offset, after: $after, before: $before, condition: $condition, filter: $filter, orderBy: $orderBy) {totalCount, pageInfo {
    hasNextPage
    hasPreviousPage
    endCursor
    startCursor
  }, nodes {
    id
    photo
    title
  }}
}
"
`;

exports[`getMany should whitelist selected fields 2`] = `"getActionsQuery"`;

exports[`getOne 1`] = `
"query getActionQuery($id: UUID!) {
  action(id: $id) {
    id
    photo
    title
  }
}
"
`;

exports[`getOne 2`] = `"getActionQuery"`;

exports[`selects all scalar fields of junction table by default 1`] = `
"query getActionGoalsQuery($first: Int, $last: Int, $after: Cursor, $before: Cursor, $offset: Int, $condition: ActionGoalCondition, $filter: ActionGoalFilter, $orderBy: [ActionGoalsOrderBy!]) {
  actionGoals(first: $first, last: $last, offset: $offset, after: $after, before: $before, condition: $condition, filter: $filter, orderBy: $orderBy) {totalCount, pageInfo {
    hasNextPage
    hasPreviousPage
    endCursor
    startCursor
  }, nodes {
    createdBy
    updatedBy
    createdAt
    updatedAt
    actionId
    goalId
  }}
}
"
`;

exports[`selects belongsTo relation field 1`] = `
"query getActionsQuery($first: Int, $last: Int, $after: Cursor, $before: Cursor, $offset: Int, $condition: ActionCondition, $filter: ActionFilter, $orderBy: [ActionsOrderBy!]) {
  actions(first: $first, last: $last, offset: $offset, after: $after, before: $before, condition: $condition, filter: $filter, orderBy: $orderBy) {totalCount, pageInfo {
    hasNextPage
    hasPreviousPage
    endCursor
    startCursor
  }, nodes {
    owner {
      id
    }
  }}
}
"
`;

exports[`selects non-scalar custom types 1`] = `"getActionsQuery"`;

exports[`selects relation field 1`] = `
"query getActionsQuery($first: Int, $last: Int, $after: Cursor, $before: Cursor, $offset: Int, $condition: ActionCondition, $filter: ActionFilter, $orderBy: [ActionsOrderBy!]) {
  actions(first: $first, last: $last, offset: $offset, after: $after, before: $before, condition: $condition, filter: $filter, orderBy: $orderBy) {totalCount, pageInfo {
    hasNextPage
    hasPreviousPage
    endCursor
    startCursor
  }, nodes {
    id
    slug
    photo
    title
    url
    goals(first: 3) {totalCount, nodes {
      id
      name
      slug
      shortName
      icon
      subHead
      tags
      search
      createdBy
      updatedBy
      createdAt
      updatedAt
    }}
  }}
}
"
`;

exports[`should select totalCount in subfields by default 1`] = `
"query getActionsQuery($first: Int, $last: Int, $after: Cursor, $before: Cursor, $offset: Int, $condition: ActionCondition, $filter: ActionFilter, $orderBy: [ActionsOrderBy!]) {
  actions(first: $first, last: $last, offset: $offset, after: $after, before: $before, condition: $condition, filter: $filter, orderBy: $orderBy) {totalCount, pageInfo {
    hasNextPage
    hasPreviousPage
    endCursor
    startCursor
  }, nodes {
    id
    photo
    title
  }}
}
"
`;

exports[`should select totalCount in subfields by default 2`] = `"getActionsQuery"`;

exports[`update with custom selection 1`] = `
"mutation updateActionMutation($id: UUID!, $slug: String, $photo: JSON, $shareImage: JSON, $title: String, $titleObjectTemplate: String, $url: String, $description: String, $discoveryHeader: String, $discoveryDescription: String, $notificationText: String, $notificationObjectTemplate: String, $enableNotifications: Boolean, $enableNotificationsText: String, $search: FullText, $location: GeoJSON, $locationRadius: BigFloat, $timeRequired: IntervalInput, $startDate: Datetime, $endDate: Datetime, $approved: Boolean, $published: Boolean, $isPrivate: Boolean, $rewardAmount: BigFloat, $activityFeedText: String, $callToAction: String, $completedActionText: String, $alreadyCompletedActionText: String, $selfVerifiable: Boolean, $isRecurring: Boolean, $recurringInterval: IntervalInput, $oncePerObject: Boolean, $minimumGroupMembers: Int, $limitedToLocation: Boolean, $tags: [String], $groupId: UUID, $ownerId: UUID, $objectTypeId: Int, $rewardId: UUID, $verifyRewardId: UUID, $photoUpload: Upload, $shareImageUpload: Upload) {
  updateAction(input: {id: $id, patch: {slug: $slug, photo: $photo, shareImage: $shareImage, title: $title, titleObjectTemplate: $titleObjectTemplate, url: $url, description: $description, discoveryHeader: $discoveryHeader, discoveryDescription: $discoveryDescription, notificationText: $notificationText, notificationObjectTemplate: $notificationObjectTemplate, enableNotifications: $enableNotifications, enableNotificationsText: $enableNotificationsText, search: $search, location: $location, locationRadius: $locationRadius, timeRequired: $timeRequired, startDate: $startDate, endDate: $endDate, approved: $approved, published: $published, isPrivate: $isPrivate, rewardAmount: $rewardAmount, activityFeedText: $activityFeedText, callToAction: $callToAction, completedActionText: $completedActionText, alreadyCompletedActionText: $alreadyCompletedActionText, selfVerifiable: $selfVerifiable, isRecurring: $isRecurring, recurringInterval: $recurringInterval, oncePerObject: $oncePerObject, minimumGroupMembers: $minimumGroupMembers, limitedToLocation: $limitedToLocation, tags: $tags, groupId: $groupId, ownerId: $ownerId, objectTypeId: $objectTypeId, rewardId: $rewardId, verifyRewardId: $verifyRewardId, photoUpload: $photoUpload, shareImageUpload: $shareImageUpload}}) {
    action {
      id
      photo
      title
    }
  }
}
"
`;

exports[`update with custom selection 2`] = `"updateActionMutation"`;

exports[`update with default scalar selection 1`] = `
"mutation updateActionMutation($id: UUID!, $slug: String, $photo: JSON, $shareImage: JSON, $title: String, $titleObjectTemplate: String, $url: String, $description: String, $discoveryHeader: String, $discoveryDescription: String, $notificationText: String, $notificationObjectTemplate: String, $enableNotifications: Boolean, $enableNotificationsText: String, $search: FullText, $location: GeoJSON, $locationRadius: BigFloat, $timeRequired: IntervalInput, $startDate: Datetime, $endDate: Datetime, $approved: Boolean, $published: Boolean, $isPrivate: Boolean, $rewardAmount: BigFloat, $activityFeedText: String, $callToAction: String, $completedActionText: String, $alreadyCompletedActionText: String, $selfVerifiable: Boolean, $isRecurring: Boolean, $recurringInterval: IntervalInput, $oncePerObject: Boolean, $minimumGroupMembers: Int, $limitedToLocation: Boolean, $tags: [String], $groupId: UUID, $ownerId: UUID, $objectTypeId: Int, $rewardId: UUID, $verifyRewardId: UUID, $photoUpload: Upload, $shareImageUpload: Upload) {
  updateAction(input: {id: $id, patch: {slug: $slug, photo: $photo, shareImage: $shareImage, title: $title, titleObjectTemplate: $titleObjectTemplate, url: $url, description: $description, discoveryHeader: $discoveryHeader, discoveryDescription: $discoveryDescription, notificationText: $notificationText, notificationObjectTemplate: $notificationObjectTemplate, enableNotifications: $enableNotifications, enableNotificationsText: $enableNotificationsText, search: $search, location: $location, locationRadius: $locationRadius, timeRequired: $timeRequired, startDate: $startDate, endDate: $endDate, approved: $approved, published: $published, isPrivate: $isPrivate, rewardAmount: $rewardAmount, activityFeedText: $activityFeedText, callToAction: $callToAction, completedActionText: $completedActionText, alreadyCompletedActionText: $alreadyCompletedActionText, selfVerifiable: $selfVerifiable, isRecurring: $isRecurring, recurringInterval: $recurringInterval, oncePerObject: $oncePerObject, minimumGroupMembers: $minimumGroupMembers, limitedToLocation: $limitedToLocation, tags: $tags, groupId: $groupId, ownerId: $ownerId, objectTypeId: $objectTypeId, rewardId: $rewardId, verifyRewardId: $verifyRewardId, photoUpload: $photoUpload, shareImageUpload: $shareImageUpload}}) {
    action {
      id
      slug
      photo
      shareImage
      title
      titleObjectTemplate
      url
      description
      discoveryHeader
      discoveryDescription
      notificationText
      notificationObjectTemplate
      enableNotifications
      enableNotificationsText
      search
      location {
        geojson
      }
      locationRadius
      timeRequired {
        days
        hours
        minutes
        months
        seconds
        years
      }
      startDate
      endDate
      approved
      published
      isPrivate
      rewardAmount
      activityFeedText
      callToAction
      completedActionText
      alreadyCompletedActionText
      selfVerifiable
      isRecurring
      recurringInterval {
        days
        hours
        minutes
        months
        seconds
        years
      }
      oncePerObject
      minimumGroupMembers
      limitedToLocation
      tags
      createdBy
      updatedBy
      createdAt
      updatedAt
    }
  }
}
"
`;

exports[`update with default scalar selection 2`] = `"updateActionMutation"`;
